programs/marginfi/src/state/liquid_insurance_fund.rs

```diff
@@ -0,0 +1,321 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{transfer, Transfer};
+use fixed::types::I80F48;
+
+use crate::{check, debug, math_error, MarginfiError, MarginfiResult};
+
+use super::marginfi_group::share_value;
+
+/// Fund that represents tokenized insurance pool backed by liquidators
+/// Uses the bank's underlying insurance vault to deposit funds
+/// Creates a mint representing the tokens managed by the lif
+#[account(zero_copy(unsafe))]
+#[repr(C)]
+#[derive(AnchorDeserialize, AnchorSerialize)]
+pub struct LiquidInsuranceFund {
+    pub group: Pubkey,
+    pub authority: Pubkey,
+
+    pub bank: Pubkey,
+    pub bank_insurance_vault: Pubkey,
+    pub bank_insurance_vault_authority: Pubkey,
```

Comment:
Ok cool -- I thought this might be the case. I thought the runtime would want to know each account being modified by the program execution? But I agree we can derive these accounts from the Bank.

Reply:
The runtime will know it because we have to explicitly pass all the addresses, its just not necessary to store them as onchain data because we can deterministically derive them